File: LLD/java/ObserverPattern/Observer/EmailAlertObserver.java
package ObserverPattern.Observer;
import ObserverPattern.Observable.StocksObservable;

public class EmailAlertObserver implements NotificationAlertObserver {
    String emailId;
    StocksObservable observable;
    public EmailAlertObserver(String emailId, StocksObservable observable) {
        this.observable = observable;
        this. emailId = emailId;
    }

    @Override
    public void update() {
        sendMail(emailId, "Product is in stock Hurry Up!");
    }
    private void sendMail(String emailId, String msg) {
        System.out.println("mail sent to : " + emailId);
    }
}

File: LLD/typescript/ObserverPattern/Observer/EmailAlertObserver.ts
import NotificationAlertObserver from "./NotificationAlertObserver";

class EmailAlertObserver implements NotificationAlertObserver {
    private emailId: string;
    private observable: any; // You can replace 'any' with a proper type

    constructor(emailId: string, observable: any) {
        this.emailId = emailId;
        this.observable = observable;
    }

    update(): void {
        this.sendEmail(this.emailId, "Product is in stock! Hurry up!");
    }

    sendEmail(emailId: string, msg: string): void {
        console.log(`Email sent to: ${emailId}`);
    }
}

export default EmailAlertObserver;
